synapse/storage/background_updates.py

```diff
@@ -249,7 +249,10 @@ async def _do_background_update(self, desired_duration_ms: float) -> int:
             retcol="progress_json",
         )
 
-        progress = json.loads(progress_json)
+        # Avoid a circular import.
+        from synapse.storage._base import db_to_json
```

Comment:
hrm. It's unsatisfactory if `synapse.storage._base` has a dependency on this file somehow. This is fine as a temporary solution though.

Reply:
`synapse.storage._base` imports from `synapse.storage.database`, which imports from `synapse.storage.background_updater`.